### DISCUSSION TITLE
RAW + JPEG Pairs

### DISCUSSION THREAD

**Discussion by gyansinha** [RhetTbull/osxphotos#581] [Q&A]
In these instances (say from a Leica with RWL and JPEG), when I export using the --skip-edited option and under the yyyy-mm-dd directory template, certain RWL files are skipped and only the JPEG saved, whereas other RWL files do get exported in the same directory.

Is this behavior expected?

**Comment by RhetTbull** [comment]
It's not expected but the handling of RAW+JPEG pairs is an area where sometimes osxphotos has to make some assumptions and doesn't always get it right. (ref #218 #225 #101 #478 ). The behavior is also different depending on whether or not the photo was missing and had to be downloaded from iCloud.  What's the full command line you used?  

If you can identify a couple of images in each category I can walk you through running a couple of commands that will gather some additional debug data that might help me track down the specific issue.

**Comment by gyansinha** [comment]
This  is the config I am using:
```
[export]
directory = "{exif.camera_make,Generic}/{exif.camera_model,Generic}/{created.year}/{created.month}/{created.dd}"
download_missing = true
jpeg_ext = "jpeg"
only_photos = true
report = "iphoto_rpt.csv"
skip_edited = true
touch_file = true
```
![Screenshot from 2022-01-10 19-24-15](https://user-images.githubusercontent.com/2814628/148860067-2ec88c76-14ae-4a48-817a-0ccf827d99ee.png)

This sub-directory shows the behavior - hope this is useful.

_P.S.: This is a superbly useful module you've written here - finally gave me the courage to tackle the crud built up over almost 2 decades an d iPhoto and Aperture transitions!_ Thanks a lot and let me know if I can contribute somehow.

**Reply by RhetTbull** [reply]
Here's a way to get some additional debug data:

First, need to find the unique IDs (UUID) of the photos in question:

`osxphotos query --name L1000594 --name L1000595 --name L1000598 --name L1000601 --name L1000602 | cut -d, -f 1`

This will print out a list of UUIDs for some of these files in each category (jpeg only / jpeg + RWL / RWL only)

The UUIDs will look something like: `4D521201-92AC-43E5-8F7C-59BC41C37A96`

Then copy the UUIDs to the following command line substituting the UUIDs for UUID1, UUID2, etc. below

`osxphotos debug-dump --dump photos --uuid UUID1 --uuid UUID2 --uuid UUID3 --uuid UUID4 --uuid UUID5 > debug.txt`

Then post the debug.txt file here.  

**Note**: The file could have sensitive info in it such as location or reverse geolocation data.  You can edit in an editor and remove any info you deem private before posting.

**Reply by gyansinha** [reply]
I did what you suggested - the debug file is attached.

[debug.txt](https://github.com/RhetTbull/osxphotos/files/7847401/debug.txt)

**Reply by RhetTbull** [reply]
Thanks! I'll take a look at the data and see what I can figure out.

**Reply by RhetTbull** [reply]
@gyansinha I've been trying to replicate this but so far unable to.  As best as I can tell from the debug data you sent the files that were exported only as JPEG didn't have an associated RAW file.  Can you look at photo `L1000594.JPG` in Photos and verify it actually has an associated RAW image?  If so, it'll have a small "R" or small "J" in top left corner (like the image below)

<img width="198" alt="Screen Shot 2022-01-13 at 10 25 24 PM" src="https://user-images.githubusercontent.com/41546558/149461597-592a0002-045b-4335-ae79-ecb31281c3ce.png">
.

**Reply by gyansinha** [reply]
You are right - when I look at my iPhoto library, I see only the JPEG version, and no raw image for L1000594. I guess somehow the raw image didn't get incorporated in the many different version upgrades along the way, or somehow I had only JPEG mode on in the camera (although unlikely since there are RWL images for that same date in the iPhoto library.

Thanks for looking into this and want to reiterate how useful this program is.

**Reply by RhetTbull** [reply]
Great! Glad it's not a bug!
